components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/mappings/SubscriptionThrottlePolicyMappingUtil.java

```diff
@@ -0,0 +1,157 @@
+                    SubscriptionThrottlePolicyDTO dto = fromSubscriptionThrottlePolicyToDTO((SubscriptionPolicy)
+                            policy);
+                    subscriptionPolicyDTOList.add(dto);
+                } else {
+                    log.error("policy object " + policy.toString() + " is not aa Subscription Policy, hence will not "
+                            + "be considered");
+                }
+            }
+        }
+        listDTO.setCount(subscriptionPolicyDTOList.size());
+        listDTO.setList(subscriptionPolicyDTOList);
+        return listDTO;
+    }
+
+    /**
+     * Converts a single Subscription Policy model into REST API DTO
+     *
+     * @param policy Subscription Policy model object
+     * @return Converted Subscription policy REST API DTO object
+     * @throws SubscriptionThrottlePolicyException
+     */
+    public static SubscriptionThrottlePolicyDTO fromSubscriptionThrottlePolicyToDTO(Policy policy)
+            throws SubscriptionThrottlePolicyException {
+        try {
+            SubscriptionThrottlePolicyDTO policyDTO = new SubscriptionThrottlePolicyDTO();
+            policyDTO = CommonThrottleMappingUtil.updateFieldsFromToPolicyToDTO(policy, policyDTO);
+            if (policy instanceof SubscriptionPolicy) {
+                SubscriptionPolicy subscriptionPolicy = (SubscriptionPolicy) policy;
+                policyDTO.setBillingPlan(subscriptionPolicy.getBillingPlan());
+                policyDTO.setRateLimitCount(subscriptionPolicy.getRateLimitCount());
+                policyDTO.setRateLimitTimeUnit(subscriptionPolicy.getRateLimitTimeUnit());
+                policyDTO.setStopOnQuotaReach(subscriptionPolicy.isStopOnQuotaReach());
+                byte[] customAttributes = subscriptionPolicy.getCustomAttributes();
+                if (customAttributes != null && customAttributes.length > 0) {
+                    List<CustomAttributeDTO> customAttributeDTOs = new ArrayList<>();
+                    JSONParser parser = new JSONParser();
+                    JSONArray attributeArray = (JSONArray) parser.parse(new String(customAttributes, StandardCharsets.UTF_8));
+                    for (Object attributeObj : attributeArray) {
+                        JSONObject attribute = (JSONObject) attributeObj;
+                        CustomAttributeDTO customAttributeDTO = CommonThrottleMappingUtil.getCustomAttribute(
+                                attribute.get(RestApiConstants.THROTTLING_CUSTOM_ATTRIBUTE_NAME).toString(),
+                                attribute.get(RestApiConstants.THROTTLING_CUSTOM_ATTRIBUTE_VALUE).toString());
+                        customAttributeDTOs.add(customAttributeDTO);
+                    }
+                    policyDTO.setCustomAttributes(customAttributeDTOs);
+                }
+            } else {
+                log.error("policy object " + policy.toString() + " is not aa Subscription Policy, hence will not "
```

Comment:
Fixed in https://github.com/wso2/carbon-apimgt/commit/8bf295d69103fd9391adef96f70d05a766cb4879.